packages/timeline-state-resolver-types/src/tricaster.ts

```diff
@@ -0,0 +1,307 @@
+import { Mapping } from './mapping'
+import { DeviceType, TimelineDatastoreReferencesContent, TSRTimelineObjBase } from '.'
+
+export type TriCasterMixEffectName = 'main' | `v${number}`
+export type TriCasterKeyerName = `dsk${number}`
+export type TriCasterInputName = `input${number}`
+export type TriCasterSourceName = TriCasterInputName | `ddr${number}` | `bfr${number}` | 'black'
+export type TriCasterAudioChannelName = TriCasterSourceName | 'sound' | 'master'
+export type TriCasterLayerName = 'a' | 'b' | 'c' | 'd'
+export type TriCasterDelegateName = 'background' | TriCasterKeyerName
+export type TriCasterMixOutputName = `mix${number}`
+export type TriCasterMixOutputSource =
+	| TriCasterSourceName
+	| TriCasterMixEffectName
+	| 'program'
+	| 'preview'
+	| 'program_clean'
+	| 'me_program'
+	| 'me_preview'
+
+interface MappingTriCasterBase extends Mapping {
+	device: DeviceType.TRICASTER
+	mappingType: MappingTriCasterType
+}
+
+export interface MappingTriCasterMixEffect extends MappingTriCasterBase {
+	mappingType: MappingTriCasterType.MixEffect
+	name: TriCasterMixEffectName
+}
+
+export interface MappingTriCasterDownStreamKeyer extends MappingTriCasterBase {
+	mappingType: MappingTriCasterType.DownStreamKeyer
+	name: TriCasterKeyerName
+}
+
+export interface MappingTriCasterInput extends MappingTriCasterBase {
+	mappingType: MappingTriCasterType.Input
+	name: TriCasterInputName
+}
+
+export interface MappingTriCasterAudioChannel extends MappingTriCasterBase {
+	mappingType: MappingTriCasterType.AudioChannel
+	name: TriCasterAudioChannelName
+}
+
+export interface MappingTriCasterMixOutput extends MappingTriCasterBase {
+	mappingType: MappingTriCasterType.MixOutput
+	name: TriCasterMixOutputName
+}
+
+export enum MappingTriCasterType {
```

Comment:
We should try to keep our formatting of our enums consistent. You have formated the attributes here using PascalCasing, but twenty lines down you have formatted them using SNAKE_CASING.
I am personally in favour of the SNAKE_CASING, but the most important part is that we try to be consistent :)

Reply:
Sidenote: Do Tricaster need the numeric value (i.e. 0, 1, 2 etc)? If not then I highly suggest that we don't assign numeric values to our enums